comment_scan/replier.rb

```diff
@@ -0,0 +1,296 @@
+require './db'
+require 'httparty'
+require 'htmlentities'
+
+class Replier
+    attr_reader :chatter, :scanner, :seclient, :BOT_NAMES
+
+    def initialize(chatter, seclient, scanner, bot_names)
+        @chatter = chatter
+        @scanner = scanner
+        @seclient = seclient
+        @BOT_NAMES = bot_names
+
+        @mc_replies = Hash.new()
+        @hg_replies = Hash.new()
+
+        @mention_actions = []
+        @fall_through_actions = []
+        @reply_actions = Hash.new()
+        @howgood_actions = Hash.new()
+
+        @mention_actions.push(method(:cat_mentions))
+
+        @fall_through_actions.push(method(:bad_command))
+
+        @reply_actions["tp"] = method(:tp)
+        @reply_actions["fp"] = method(:fp)
+        @reply_actions["wrongo"] = method(:fp) #fun fp alias
+        @reply_actions["rude"] = method(:rude)
+        @reply_actions["dbid"] = method(:dbid)
+        @reply_actions["feedbacks"] = method(:feedbacks)
+        @reply_actions["del"] = method(:del_reply)
+        @reply_actions["huh?"] = method(:huh)
+        @reply_actions["rescan"] = method(:rescan)
+        @reply_actions["report"] = method(:report)
+
+        @howgood_actions["tp"] = method(:howgood_tp) #For tp responses to howgood
+        @howgood_actions["fp"] = method(:howgood_fp) #For fp responses to howgood
+        @howgood_actions["*"] = method(:howgood_glob) #For * responses to howgood
+    end
+
+    def setup_reply_actions()
+        @reply_actions.each do |command, action|
+            @chatter.add_reply_action(command, action, [self])
+        end
+
+        @howgood_actions.each do |command, action|
+            @chatter.add_reply_action(command, action, [self])
+        end
+    end
+
+    def setup_mention_actions()
+        @mention_actions.each do |action|
+            @chatter.add_mention_action(action, [self])
+        end
+    end
+
+    def setup_fall_through_actions()
+        @fall_through_actions.each do |action|
+            @chatter.add_fall_through_reply_action(action, [self])
+        end
+    end
+
+    def show_first_n_comments(db_comments, num_to_display)
+        # Pull comment_id's from the first num_to_display comments we matched to pass to scan
+        db_comments.take(num_to_display).each { |comment| @scanner.repot_db_comment(comment, should_post_matches: true) }
+    end
+
+    def random_response()
+      responses = ["Ain't that the truth.",
+                   "You're telling me.",
+                   "Yep. That's about the size of it.",
+                   "That's what I've been saying for $(AGE_OF_BOT)!",
+                   "What else is new?",
+                   "For real?",
+                   "Humans, amirite?"]
+
+      return responses[rand(responses.length())]
+    end
+
+    def contains_cat(message)
+        ["cat", "kitty", "kitties", "kitten", "meow", "purr", "feline"].any? { |cat_name| message.downcase.include? cat_name }
+    end
+end
+
```

Comment:
Why aren't these below methods in any class?

Reply:
I guess I just assumed function pointers would work like C++, where you only use a function pointer to a class method if you really hate yourself (and then things get crazy complicated). I just tested it though and it looks like Ruby realizes it's a class method and passes a copy of `self` with the function pointer all for you behind the scenes. So I can definitely clean a lot of this up (and avoid passing `payload` around).